QuadPaySDK/Widget/Widget.swift

```diff
@@ -11,24 +11,39 @@ import UIKit
 
 public final class Widget : UIView{
     
+    private var request: ApiRequest<WidgetDataResource>?
+    
     @objc public var merchantId: String = ""{
         didSet{
-            MerchantService.shared.fetchMerchants(merchantId: merchantId){
-                (result) in
-                switch result {
-                case .success(_):
-                    self.grayLabelMerchant = true
-                    DispatchQueue.main.async {
-                        self.layout()
+            if(amount != "0"){
```

Comment:
This will suffer from the same problem if the `merchantId` is set first before the `amount`, no widget data will be fetched.